### PR TITLE
Fixed issue #69 

### PR DESCRIPTION
**Pull request by yixiang**
Fixed issue #69. Indentation resets when formatting selected range.

### CODE REVIEW

**Comment by yixiang** [comment]
Now the text selection after formatting is correct.

**Comment by tonyarnold** [comment]
It's much better, but still having some issues:

screen shot 2015-03-25 at 10 13 17

After formatting:

screen shot 2015-03-25 at 10 13 29

Also, you don't need to open a new PR to make changes — just push any updates to your `indent` branch and this PR will automatically include them.

File: ClangFormat/TRVSCodeFragment.m [hunk 1]
```diff
@@ -34,14 +34,57 @@ + (instancetype)fragmentUsingBlock:
...
+- (void)updateRangeToReplace:(NSString *)formattedDoc {
+  NSUInteger originalLen = _string.length;
+  NSUInteger formattedLen = formattedDoc.length;
+  NSRange originalSelection = _textRangePreFormat;
+  NSRange rangeInOriginalDoc = NSMakeRange(0, _string.length);
+  NSRange rangeInFormattedDoc = NSMakeRange(0, formattedDoc.length);
+
+  // Find the left boundary.
+  NSUInteger i;
+  for (i = 0; i < _textRangePreFormat.location - 1; ++i) {
```
Comment (tonyarnold) [inline ClangFormat/TRVSCodeFragment.m]: It looks like the `_textRangePreFormat.location - 1` is the culprit behind the issues I posted to this PR. In my (limited) local tests, removing the `-1` makes this code behave as expected.

**Comment by yixiang** [comment]
I've fixed the one-off error you pointed out. I think it's pretty solid now.

**Comment by tonyarnold** [comment]
:+1: agreed. Thanks for resolving the problems! /cc @travisjeffery

**Comment by travisjeffery** [comment]
:+1: Thanks fellas!

**Pull request merged**